#1007 feat(route) route first frame of layer background producer

Julusian opened the pull request:
```
PLAY 2-10 route://1-10 BACKGROUND
PLAY 2-10 route://1-10 NEXT
```
This allows you to view the first frame of the background producer. Intended as a preview of the source to make it possible to visibly show to users that the clip is loaded and it is the correct clip.
BACKGROUND mode will only ever show the first frame of the layer background producer,
NEXT mode will show the first frame of the background clip if one is loaded, otherwise will show the foreground. So if you LOADBG then PLAY, the clip will continue playing.

@ronag I have tried to avoid calling receive on the background producers as much as possible, to avoid adding additional cpu usage for who doesn't use this feature. It would be a bit cleaner to always generate them though.

This also fixes a bug where CLEARing a layer which was being routed would cause that route to freeze on last frame until any operation was performed on that empty layer (even a CALL would cause it to unfreeze). This was from the layer no longer existing, so there was no frame to send to a route, when we should have been sending an empty frame.

Julusian commented:
> @ronag I don't understand what exactly this is fixing/doing?

Sorry, I missed out a chunk of the description. hopefully that helps explain the purpose?

TomKaltz commented:
This should also allow OSC to get background producer information correct?

Julusian commented:
> @TomKaltz This should also allow OSC to get background producer information correct?

If we always generate the first frame of any background producer then it should give us all the osc data wanted and it would be simple to then expose that. 
I didn't want to do that as part of this as I'm not 100% certain of the performance implications of always generating the first frame of the background producers, or if there would be any other issues arising from doing so

premultiply commented:
There is one thing to make shure when the background producers first frame is brought to the foreground: The first frame and audio has to be played again.

And maybe the "first frame" implementation is even usefull for generating thumbnail files for any producer/content.

Julusian commented:
@premultiply good point. I havent checked that, and suspect this code will not handle that correctly

TomKaltz commented:
meta #941

Julusian commented:
I have rebased and reworked this a little with some fixes.

I can confirm that the first frame is still played on the output after being shown via a background route.

Additionally, any real-time sources (decklink, html etc) will now show motion, as the route shows the first frame of their buffers. This is to ensure it is a true representation of what will be shown on PLAY rather than the true first frame of the producer which is likely very different.

The transition will always show the first frame of the destination producer instead of the transition, as the transition will almost always start with showing just the source
